### PR TITLE
Mobile, menu fixes.  Auto-gen for version menu research

### PR DESCRIPTION
**Pull request by Krief**
The version menu should use the docsversion variable from the solo site variables.  However, the generation is tough to test currently and will require more changes, with some testing in the live environment, all of which needs to take a back-burner to more pressing issues.

A path is clearly laid out to change versionnavigation.html to an appropriate {{ range }} and test away.

### CODE REVIEW

File: layouts/partials/versionnavigation.html [hunk 1]
```diff
@@ -1,6 +1,15 @@
...
+            text-overflow: ellipsis;
+            white-space: nowrap;"
+        >
+          {{ .Site.Data.Solo.CodeVersion }}
+        </div>-->
+  
+<!-- Need to run throuugh a RANGE of .Site.Data.Solo.DocsVersions .  This is tough to test locally as the version folders aren't generated -->
+<label for="version-selection" id="version-selection-label">Version:</label>
+<select id="version-selection" onchange="javascript:location.href = this.value;">
+  <option value="/latest/{{ .RelPermalink}}" {{if in .RelPermalink "latest/"}} selected{{end}}>
```
Comment (kdorosh) [inline layouts/partials/versionnavigation.html]: i dont' think we want the trailing slash, i.e.:
```suggestion
  <option value="/latest{{ .RelPermalink}}" {{if in .RelPermalink "latest/"}} selected{{end}}>
```

Comment (Krief) [reply layouts/partials/versionnavigation.html]: I think we do?  And just tested it live (as this isn't really new code. More just moved around).

Comment (kdorosh) [reply layouts/partials/versionnavigation.html]: This is the version currently deployed: https://github.com/solo-io/hugo-theme-soloio/blob/master/layouts/partials/flex/versionnavigation.html#L3 , which works as expected and doesn't render an extra `/`

Comment (Krief) [reply layouts/partials/versionnavigation.html]: that file wasn't actually previously being used.

Comment (kdorosh) [reply layouts/partials/versionnavigation.html]: oops this is the one i meant to link: https://github.com/solo-io/hugo-theme-soloio/blob/master/layouts/partials/flex/body-beforecontent.html#L23

File: layouts/partials/versionnavigation.html [hunk 2]
```diff
@@ -1,6 +1,15 @@
...
+            text-overflow: ellipsis;
+            white-space: nowrap;"
+        >
+          {{ .Site.Data.Solo.CodeVersion }}
+        </div>-->
+  
+<!-- Need to run throuugh a RANGE of .Site.Data.Solo.DocsVersions .  This is tough to test locally as the version folders aren't generated -->
+<label for="version-selection" id="version-selection-label">Version:</label>
+<select id="version-selection" onchange="javascript:location.href = this.value;">
+  <option value="/latest{{ .RelPermalink}}" {{if in .RelPermalink "latest/"}} selected{{end}}>
```
Comment (kdorosh) [inline layouts/partials/versionnavigation.html]: needs to be `/gloo/latest`

**Pull request merged**